torch/sparse/__init__.py

```diff
@@ -525,3 +525,178 @@ def func_wrapper(*args, **kwargs):
+
+                        if values.numel() > 0:
+                            strides = torch.empty(obj.shape[batch_dim:batch_dim + 2]).stride()
+                            if batch_dim > 0:
+                                batch_compressed_indices = compressed_indices.view(-1, *compressed_indices.shape[batch_dim:])
+                                batch_plain_indices = plain_indices.view(-1, *plain_indices.shape[batch_dim:])
+                                batch_values = values.view(-1, *values.shape[batch_dim:])
+                                batch_full_values = full_values.view(-1, *full_values.shape[batch_dim:])
+                                for i in range(batch_shape.numel()):
+                                    # TODO: eliminate this for-loop after gh-104868 is fixed
+                                    indices = torch._convert_indices_from_csr_to_coo(
+                                        batch_compressed_indices[i], batch_plain_indices[i])
+                                    flatten_indices = (torch.tensor([strides], device=device, dtype=indices.dtype).T
+                                                       * indices).sum(0)
+                                    batch_full_values[i][flatten_indices] = batch_values[i]
+                            else:
+                                indices = torch._convert_indices_from_csr_to_coo(compressed_indices, plain_indices)
+                                flatten_indices = (torch.tensor([strides], device=device, dtype=indices.dtype).T * indices).sum(0)
+                                full_values[flatten_indices] = values
+
+                        full_indices = torch.ones(obj.shape[:batch_dim + 2],
+                                                  device=device, dtype=torch.int8).nonzero().to(dtype=torch.int64).T
+                        d.update(
+                            indices=full_indices,  # TODO: eliminate full_indices after gh-107126 is fixed
+                            compressed_indices=full_compressed_indices,
+                            plain_indices=full_plain_indices)
+                        return (STRIDED_REPRESENTATION, d, full_values.requires_grad_(True))
+                elif obj.layout in {torch.sparse_bsr, torch.sparse_csc, torch.sparse_bsc}:
+                    raise NotImplementedError(f'converstion of {obj.layout} tensor to strided representation')
+                else:
+                    return obj
+            return obj
+
+        def _restore_from_strided_representation(d, values):
+            """Restore a non-strided differentiable tensor from its strided
+            representation.
+            """
+            if d['layout'] is torch.sparse_coo:
+                # TODO: After fixing gh-107097, replace `.coalesce()`
+                # method call with `._coalesced_(True)`.
+                return torch.sparse_coo_tensor(d['indices'], values, size=d['shape']).coalesce()
```

Comment:
Ideally, it should be `._coalesced_(d['original'].is_coalesced())` but using this is blocked by https://github.com/pytorch/pytorch/issues/107097.

Otherwise, using `coalesce()` is not required here indeed.

Reply:
Ok!

Reply:
coalesce is removed, looking forward to https://github.com/pytorch/pytorch/issues/107097 fix.